### PR TITLE
Chardonnay Bugfix + Moved Default Settings into Configuration

### PR DESCRIPTION
**Pull request by Mbucari**
Hey, sorry for the rocky rollout of 8.8.

This PR is really to fix a couple of chardonnay bugs that I just noticed, but it also contains the migration of default settings into Configuration because I had already committed those.

I know 17 files looks like a lot, but 10 of them are <= 3 lines of changes.

I tested all settings with an existing settings file and a fresh installation, and every setting works as expected.

### CODE REVIEW

File: Source/LibationFileManager/Configuration.cs [hunk 1]
```diff
@@ -24,9 +24,6 @@ public static bool SettingsFileIsValid(string settingsFile)
 			if (booksDir is null || !Directory.Exists(booksDir))
 				return false;
 
-			if (string.IsNullOrWhiteSpace(pDic.GetString(nameof(InProgress))))
-				return false;
-
```
Comment (Mbucari) [inline Source/LibationFileManager/Configuration.cs:29]: IsNullOrWhiteSpace check moved into InProgress getter. Returns WinTemp if null or white space.

**Comment by Mbucari** [comment]
I've tested the crap out of both classic and chardonnay, and everything I've tried works as expected.

File: Source/AppScaffolding/LibationScaffolding.cs [hunk 2]
```diff
@@ -86,110 +85,6 @@ public static void RunPostConfigMigrations(Configuration config)
...
 		}
 
-		public static void PopulateMissingConfigValues(Configuration config)
-		{
-			config.InProgress ??= Configuration.WinTemp;
-
-            if (!config.Exists(nameof(config.UseCoverAsFolderIcon)))
-                config.UseCoverAsFolderIcon = false;
-
-            if (!config.Exists(nameof(config.BetaOptIn)))
```
Comment (rmcrackan) [inline Source/AppScaffolding/LibationScaffolding.cs:96]: It feels like the right move to get these values out of scaffolding and into their own domain areas. I didn't check each of these default throughout your PR but I trust that you did

File: Source/LibationFileManager/Configuration.PersistentSettings.cs [hunk 3]
```diff
@@ -20,9 +20,9 @@ public partial class Configuration
 
 		private PersistentDictionary persistentDictionary;
 
-		public T GetNonString<T>([CallerMemberName] string propertyName = "") => persistentDictionary.GetNonString<T>(propertyName);
+		public T GetNonString<T>(T defaultValue, [CallerMemberName] string propertyName = "") => persistentDictionary.GetNonString(propertyName, defaultValue);
 		public object GetObject([CallerMemberName] string propertyName = "") => persistentDictionary.GetObject(propertyName);
-		public string GetString([CallerMemberName] string propertyName = "") => persistentDictionary.GetString(propertyName);
+		public string GetString(string defaultValue = null, [CallerMemberName] string propertyName = "") => persistentDictionary.GetString(propertyName, defaultValue);
```
[the remark this answers is not in the document]
Comment (rmcrackan) [reply Source/LibationFileManager/Configuration.PersistentSettings.cs:25]: What happens when the value is changed from default, to non default, then back to default? Is the entry removed from persistence? Or is the default value written? Either is fine. The only wrong answer is if the new value is detected to be default and persistence thinks it has no action -- thus leaving the non-default value in place

**Comment by rmcrackan** [comment]
Reviewed. It feels weird but right.

Just that one question about default then I'm ready to merge.

File: Source/AppScaffolding/LibationScaffolding.cs (hunk 2 above)
Comment (Mbucari) [reply Source/AppScaffolding/LibationScaffolding.cs:96]: I did. One by one I deleted them from scaffolding and added the default value into configuration.

**Comment by Mbucari** [comment]
If you decide you want default settings to be written to the Jason file, you can just edit PersistentDictionary to write the default value when the property is not found.

File: Source/LibationFileManager/Configuration.PersistentSettings.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmcrackan) [reply Source/LibationFileManager/Configuration.PersistentSettings.cs:25]: I just tried this. It persists the default, which is fine.

**Pull request merged**